src/hubbleds/tools/bin_select.py

```diff
@@ -0,0 +1,144 @@
+from glue_jupyter.bqplot.common.tools import BqplotSelectionTool
+from glue.config import viewer_tool
+from echo import CallbackProperty
+from bqplot_image_gl.interacts import MouseInteraction, mouse_events
+from glue_jupyter.bqplot.common.tools import InteractCheckableTool, CheckableTool
+
+
+from glue.config import viewer_tool
+from glue_jupyter.bqplot.common.tools import INTERACT_COLOR
+
+from contextlib import nullcontext
+
+
+from bqplot.interacts import BrushIntervalSelector, IndexSelector
+
+from glue.core.roi import RangeROI
+from glue.core.subset import RangeSubsetState
+from glue.config import viewer_tool
+import numpy as np
+
+
+# this decorator tells glue this is a viewer tool, so it knows what to do with
+# all this info
+@viewer_tool
+class BinSelect(BqplotSelectionTool):
+    icon = 'glue_xrange_select'
+    mdi_icon = "mdi-select-compare"
+    tool_id = 'hubble:binselect'
+    action_text = 'Select fully enclosed bins'
+    tool_tip = 'Select fully enclosed bins'
+    tool_activated = CallbackProperty(False)
+    x_min = CallbackProperty(0)
+    x_max = CallbackProperty(0)
+    
+
+    def __init__(self, viewer, **kwargs):
+
+        super().__init__(viewer, **kwargs)
+
+        self.interact = BrushIntervalSelector(scale=self.viewer.scale_x,
+                                              color=INTERACT_COLOR)
+
+        self.interact.observe(self.update_selection, "brushing")
+
+    def update_selection(self, *args):
+        with self.viewer._output_widget or nullcontext():
```

Comment:
Just curious, why do this and `activate` have context managers? The purpose isn't immediately clear to me.